Source/ACE/Entity/WorldObject.cs

```diff
@@ -122,156 +122,45 @@ public void SetMotionState(MotionState motionState)
             p.Session.Network.EnqueueSend(updateMotion);
         }
 
-        public void AddToInventory(WorldObject worldObject)
-        {
-            lock (this.inventoryMutex)
-            {
-                if (inventory.ContainsKey(worldObject.Guid))
-                    return;
-
-                inventory.Add(worldObject.Guid, worldObject);
-            }
-        }
-
-        public void RemoveFromInventory(ObjectGuid objectGuid)
+        // Inventory Management Functions
+        public virtual void AddToInventory(WorldObject inventoryItem)
         {
             lock (inventoryMutex)
             {
-                if (inventory.ContainsKey(objectGuid))
-                    inventory.Remove(objectGuid);
+                if (!inventory.ContainsKey(inventoryItem.Guid)) inventory.Add(inventoryItem.Guid, inventoryItem);
```

Comment:
Ok got rid of inline if's, think I fixed the whitespace - I may need remedial training on that to make sure I understand our style.   I fixed my last bug and think it is all ready to go.